python/element_array_view.cpp

```diff
@@ -42,6 +44,32 @@ void declare_ElementArrayView(py::module &m, const std::string &suffix) {
   });
 }
 
+template <>
+void declare_ElementArrayView<scipp::python::PyObject>(
+    py::module &m, const std::string &suffix) {
+  using T = scipp::python::PyObject;
+  py::class_<ElementArrayView<T>>(
+      m, (std::string("ElementArrayView_") + suffix).c_str())
+      .def(
+          "__repr__",
+          [](const ElementArrayView<T> &self) { return array_to_string(self); })
+      .def(
+          "__getitem__",
+          [](const ElementArrayView<T> &self, const scipp::index i) {
+            return self[i].to_pybind();
+          },
+          py::return_value_policy::reference)
+      .def(
+          "__setitem__",
+          [](ElementArrayView<T> &self, const scipp::index i,
+             const py::object &value) { return self[i].to_pybind() = value; },
+          py::return_value_policy::reference)
+      .def("__len__", &ElementArrayView<T>::size)
+      .def("__iter__", [](const ElementArrayView<T> &self) {
+        return py::make_iterator(self.begin(), self.end());
+      });
+}
```

Comment:
Quite some duplication, would it be briefer and more maintainable to use `if constexpr(std::is_same_v<T, PyObject>)` in the original template definition instead?